core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/Checksummer.java

```diff
@@ -15,15 +15,28 @@
 
 package software.amazon.awssdk.http.auth.aws.signer;
 
+import static software.amazon.awssdk.checksums.DefaultChecksumAlgorithm.SHA256;
+import static software.amazon.awssdk.http.auth.aws.internal.util.ChecksumUtil.ConstantChecksumAlgorithm;
+import static software.amazon.awssdk.http.auth.aws.internal.util.ChecksumUtil.checksumHeaderName;
+import static software.amazon.awssdk.http.auth.aws.util.SignerConstant.X_AMZ_CONTENT_SHA256;
+
 import java.nio.ByteBuffer;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import org.reactivestreams.Publisher;
 import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.checksums.spi.ChecksumAlgorithm;
 import software.amazon.awssdk.http.ContentStreamProvider;
+import software.amazon.awssdk.http.SdkHttpRequest;
 import software.amazon.awssdk.http.auth.aws.internal.signer.DefaultChecksummer;
+import software.amazon.awssdk.http.auth.aws.internal.signer.FlexibleChecksummer;
+import software.amazon.awssdk.http.auth.aws.internal.signer.PrecomputedChecksummer;
+import software.amazon.awssdk.utils.ImmutableMap;
 
 /**
  * An interface for defining how a checksum is formed from a payload synchronously and asynchronously.
+ * <p>
+ * The implementation may choose to also manipulate the request with the checksum, such as adding it as a header.
  */
 @SdkProtectedApi
```

Comment:
This can be internal, but we can make that change when we clean up protected surface.

Reply:
Yes, should we add a TODO to remember?